### PR TITLE
Use custom service accounts instead of default for controllers

### LINKED ISSUE #493: Use custom service accounts instead of default for controllers
/kind feature

A dedicated service account for each CAPG controller is more likely to be used by an operator instead of the default.

See:- 
- [kubernetes-sigs/cluster-api/issues/4015](https://github.com/kubernetes-sigs/cluster-api/issues/4015)
- [kubernetes-sigs/cluster-api-provider-azure/issues/1206](https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/1206)

### PR DESCRIPTION
**Pull request by jayesh-srivastava**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Uses a dedicated service account for each CAPG controller instead of the default.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #493 

**Special notes for your reviewer**:

**TODOs**:
- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:
```
Use custom service accounts instead of default for controllers.
```

### CODE REVIEW

**Comment by jayesh-srivastava** [comment]
cc @cpanato

File: config/rbac/service_account.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: manager
+  namespace: system
```
Comment (cpanato) [inline config/rbac/service_account.yaml:5]: nit: new line

Comment (jayesh-srivastava) [reply config/rbac/service_account.yaml]: done.

File: config/rbac/service_account.yaml [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+
```
Comment (cpanato) [inline config/rbac/service_account.yaml:6]: there is now a blank line and it is missing the new line in the end

Comment (jayesh-srivastava) [reply config/rbac/service_account.yaml]: @cpanato can you check now please?

**Comment by jayesh-srivastava** [comment]
@cpanato Yesterday all these tests passed but today these two failed. Can you help me with this?

**Comment by cpanato** [comment]
ahh, the issue is the jobs are running using go 1.17 now we need to merge https://github.com/kubernetes-sigs/cluster-api-provider-gcp/pull/501 first now that is a blocking to release main branch again

**Comment by cpanato** [comment]
@jayesh-srivastava looks like now we have an issue 

```
msg: "failed to execute kustomize: Error: accumulating resources: recursed accumulation of path '../rbac': accumulating resources: accumulating resources from 'service_account.yaml': missing metadata.name in object {map[apiVersion:v1 kind:ServiceAccount metadata:[map[name:manager namespace:system]]]}\n",
```

can you debug and see what is missing?

**Comment by jayesh-srivastava** [comment]
@cpanato Done!

> @jayesh-srivastava looks like now we have an issue
> 
> ```
> msg: "failed to execute kustomize: Error: accumulating resources: recursed accumulation of path '../rbac': accumulating resources: accumulating resources from 'service_account.yaml': missing metadata.name in object {map[apiVersion:v1 kind:ServiceAccount metadata:[map[name:manager namespace:system]]]}\n",
> ```
> 
> can you debug and see what is missing?

**Review by cpanato — APPROVED** [review]
cool, thanks @jayesh-srivastava 
/approve
/lgtm

**Pull request merged**